src/sage/combinat/lp_algebra_seed.py

```diff
@@ -0,0 +1,1311 @@
+from sage.rings.fraction_field import FractionField
+from sage.rings.infinity import infinity
+from sage.rings.integer_ring import IntegerRing_class
+from sage.rings.integer_ring import ZZ
+from sage.rings.polynomial.polynomial_ring_constructor import PolynomialRing
+from sage.rings.rational_field import RationalField
+from sage.structure.sage_object import SageObject
+from sage.misc.latex import latex
+from sage.symbolic.ring import SR
+
+
+class LPASeed(SageObject):
+    r"""
+    Initialise a Laurent phenomenon algebra seed.
+
+    INPUT:
+
+    - ``data`` -- can be one of the following:
+
+        * dict - dictionary of initial variable names to their
+            corresponding exchange polynomials
+        * LPASeed object
+
+    - ``coefficients`` -- tuple of symbolic variables (default: ``()``); the
+        labels of, if any, the coefficients of the exchange polynomials. If no
+        coefficients are provided, the module attempts to detect them from the
+        input data.
+
+    - ``base_ring`` -- unique factorisation domain (default: ``ZZ``);
+        the ring which we take the exchange polynomials over; currently
+        only supports ``ZZ`` or ``QQ``
+
+    EXAMPLES:
+
+    This example initialises a linear Laurent phenomenon algebra in two
+    variables::
+
+        sage: var('x1,x2')
+        (x1, x2)
+        sage: S = LPASeed({x1: 1 + x2, x2: 1 + x1})
+        sage: S
+        A seed with cluster variables [x1, x2] and exchange polynomials [x2 + 1, x1 + 1]
+
+    We add some coefficients to get the generic linear LP algebra in three
+    variables::
+
+        sage: var('x1,x2,x3,a0,a2,a3,b0,b1,b3,c0,c1,c2')
+        (x1, x2, x3, a0, a2, a3, b0, b1, b3, c0, c1, c2)
+        sage: S = LPASeed({x1: a0 + a2*x2 + a3*x3, x2: b0 + b1*x1 + b3*x3, x3: c0 + c1*x1 + c2*x2}, coefficients=[a0,a2,a3,b0,b1,b3,c0,c1,c2],base_ring=ZZ)
+        sage: S
+        A seed with cluster variables [x1, x2, x3] and exchange polynomials [x2*a2 + x3*a3 + a0, x1*b1 + x3*b3 + b0, x1*c1 + x2*c2 + c0]
+
+    More complicated polynomials are allowed, as long as they are
+    irreducible::
+
+        sage: var('x1,x2,x3')
+        (x1, x2, x3)
+        sage: S = LPASeed({x1: 1 + x2*x3^2 + 4*x3^3, x2: 2 - x1^2, x3: 4 + x1^3*x2^2 - 3*x1})
+        sage: S
+        A seed with cluster variables [x1, x2, x3] and exchange polynomials [x2*x3^2 + 4*x3^3 + 1, -x1^2 + 2, x1^3*x2^2 - 3*x1 + 4]
+
+    Nonirreducible polynomials will raise an exception::
+
+        sage: var('x1, x2')
+        (x1, x2)
+        sage: S = LPASeed({x1: 4 - x2^2, x2: 1 + x1})
+        Traceback (most recent call last):
+        ...
+        ValueError: (LP2) fail: -x2^2 + 4 is not irreducible over Integer Ring
+
+    Different base rings are allowed::
+
+        sage: var('x1,x2')
+        (x1, x2)
+        sage: S = LPASeed({x1: 1 + x2, x2: 1 + x1}, base_ring=QQ)
+        sage: S
+        A seed with cluster variables [x1, x2] and exchange polynomials [x2 + 1, x1 + 1]
+    """
+
```

Comment:
```suggestion
    """
```
We typically don't have this blankline here.